client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx

```diff
@@ -0,0 +1,28 @@
+import { useDispatch } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect } from 'react';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import type { Step } from '../../types';
+import './style.scss';
+
+const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
+
+const CompletingPurchase: Step = function CompletingPurchase( { navigation } ) {
+	const { __ } = useI18n();
+	const { submit } = navigation;
+	const { setPendingAction, setProgressTitle, setProgress } = useDispatch( ONBOARD_STORE );
+
+	useEffect( () => {
+		setPendingAction( async () => {
+			setProgressTitle( __( 'Completing Purchase' ) );
+			setProgress( 1 );
+			await wait( 2000 );
```

Comment:
Maybe we could add some wait before with progress at something like 0.3?
Designs ( BcUdhLlhAp7UHtQHNTI4zA-fi-3008%3A65983 ) show something like that